internal/backends/mcu/pico_st7789.rs

```diff
@@ -160,31 +202,31 @@ mod xpt2046 {
     use embedded_time::rate::Extensions;
     use euclid::default::Point2D;
 
-    pub struct XPT2046<IRQ: InputPin, CS: OutputPin, SPI: Transfer<u8>> {
-        irq: IRQ,
+    pub struct XPT2046<CS: OutputPin, SPI: Transfer<u8>> {
         cs: CS,
         spi: SPI,
         pressed: bool,
     }
 
-    impl<PinE, IRQ: InputPin<Error = PinE>, CS: OutputPin<Error = PinE>, SPI: Transfer<u8>>
-        XPT2046<IRQ, CS, SPI>
-    {
-        pub fn new(irq: IRQ, mut cs: CS, spi: SPI) -> Result<Self, PinE> {
+    impl<PinE, CS: OutputPin<Error = PinE>, SPI: Transfer<u8>> XPT2046<CS, SPI> {
+        pub fn new(mut cs: CS, spi: SPI) -> Result<Self, PinE> {
             cs.set_high()?;
-            Ok(Self { irq, cs, spi, pressed: false })
+            Ok(Self { cs, spi, pressed: false })
         }
 
         pub fn read(&mut self) -> Result<Option<Point2D<f32>>, Error<PinE, SPI::Error>> {
             const PRESS_THRESHOLD: i32 = -20_000;
             const RELEASE_THRESHOLD: i32 = -30_000;
             let threshold = if self.pressed { RELEASE_THRESHOLD } else { PRESS_THRESHOLD };
             self.pressed = false;
-            if self.irq.is_low().map_err(|e| Error::Pin(e))? {
+
+            if cortex_m::interrupt::free(|cs| {
+                super::IRQ_PIN.borrow(cs).borrow_mut().as_ref().unwrap().is_low().unwrap()
+            }) {
                 const CMD_X_READ: u8 = 0b10010000;
                 const CMD_Y_READ: u8 = 0b11010000;
-                const CMD_Z1_READ: u8 = 0b10110001;
-                const CMD_Z2_READ: u8 = 0b11000001;
+                const CMD_Z1_READ: u8 = 0b10110000;
```

Comment:
The zero in the LSB makes sure the IRQ pin stays enabled. A 1 in LSB disables it.